### DISCUSSION TITLE
Error with a basic example using React Navigation 7 : Element type is invalid: expected a string

### DISCUSSION THREAD

**Discussion by GaylordP** [react-navigation/react-navigation#12195] [Q&A] [answered]
Hello and thank you for your wonderful library, I have been using React Navigation since version 5. :)
I have already asked the question on Stackoverflow without getting a response :/

I’m trying to create a local project with the latest version of React Navigation (7RC), but even for a very simple example, I keep getting this error:

> Warning: Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports

    // In App.js in a new project

    import * as React from 'react';
    import { View, Text } from 'react-native';
    import { NavigationContainer } from '@react-navigation/native';
    import { createNativeStackNavigator } from '@react-navigation/native-stack';

    function HomeScreen() {
      return (
        <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
          <Text>Home Screen</Text>
        </View>
      );
    }

    const Stack = createNativeStackNavigator();

    function RootStack() {
      return (
        <Stack.Navigator>
          <Stack.Screen name="Home" component={HomeScreen} />
        </Stack.Navigator>
      );
    }

    export default function App() {
      return (
        <NavigationContainer>
          <RootStack />
        </NavigationContainer>
      );
    }

When I run the same very simple example ([from the react navigation documentation][1]) on Snack (to make sure the issue isn't coming from my configuration.), the error is the same : https://snack.expo.dev/6V2VhVWWhgJq3N3t767te (Select the Android version).

I am aware that this is an RC version (and I have tested several previous ones), but I’m surprised that it crashes right from the first lines of code.
Do you have any idea why? Thanks.

  [1]: https://reactnavigation.org/docs/7.x/hello-react-navigation#creating-a-native-stack-navigator

**Comment by satya164** [comment, marked as answer]
React Navigation 7's Native Stack requires React Native Screens v4 (currently in beta)

**Comment by GaylordP** [comment]
Thank you very much, that was the reason :)

**Comment by mmvsk** [comment]
And react-native-screens (4.0.0-beta.0 .. 4.0.0-beta.13) crashes:

> No ViewManager found for class RNSScreenContentWrapper

(fresh `create expo-app@latest` install, tested on Android using Expo Go)

Current workaround:

- keep native-stack in RC .27 `@react-navigation/native-stack@7.0.0-rc.27`
- use latest 3.x react-native-screens
